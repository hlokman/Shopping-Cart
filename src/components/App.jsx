import "../style/App.css";
import { Link } from "react-router-dom";

function App() {
  return (
    <>
      <header className="flex justify-around items-center bg-white p-4">
        <h1 className="text-2xl font-courierbold ">Great Shop</h1>
        <nav className="flex gap-10 font-primary text-lg">
          <Link to="/" className="">
            Home
          </Link>
          <Link to="/" className="">
            Shop
          </Link>
          <Link to="/" className="">
            About
          </Link>
        </nav>
        <div>
          <button className="bg-cart-logo bg-cover h-8 w-8 border-none pointer-events-auto"></button>
        </div>
      </header>

      <main className="bg-sky-img bg-no-repeat bg-cover min-h-[calc(100vh-71px)] opacity-95 flex justify-center items-center content-center gap-10 flex-wrap  ">
        <div className="max-w-[370px] flex flex-col gap-3 ">
          <h1 className="text-4xl font-bold text-white font-courierbold">
            The Greatest Shop
          </h1>
          <p className="text-white font-primary leading-4">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc
            pulvinar nibh arcu, sed auctor erat maximus sit amet.
          </p>
          <button className="bg-white rounded font-bold font-courierbold">
            Go Shopping
          </button>
        </div>
        <div className="bg-suit-img bg-no-repeat bg-cover w-[370px] min-h-[320px] rounded-lg shadow-[0px_0px_14px_black]"></div>
      </main>
    </>
  );
}

export default App;
