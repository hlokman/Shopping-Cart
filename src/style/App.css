body {
  margin: 0;
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  /*background-color: rgb(200, 200, 200);*/
  /*background-color: #e7e7e7;*/
}

/*button {
  background-color: inherit;
  border: none;
}

header {
  display: flex;
}

.pagesButtons {
  display: flex;
  gap: 10px;
}
.pagesButtons > button {
  font-size: 17px;
  font-weight: 700;
}

.cartLogo {
  background-image: url("../assets/cart.svg");
}*/
