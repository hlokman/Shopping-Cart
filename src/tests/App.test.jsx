import { render, screen } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router-dom";

import App from "../components/App";
import { describe, expect } from "vitest";

/*describe("App", () => {
  it("renders headline", () => {
    render(<App title="React" />);

    screen.debug();

    // check if App components renders headline
  });
});*/

describe("App component", () => {
  it("render ", () => {
    const routes = [
      {
        path: "/",
        element: <App />,
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ["/"],
      initialIndex: 0,
    });

    const { container } = render(<RouterProvider router={router} />);

    expect(container).toMatchSnapshot();
  });
});
