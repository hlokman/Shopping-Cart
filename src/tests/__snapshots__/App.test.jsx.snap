// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > render  1`] = `
<div>
  <header
    class="flex justify-around items-center bg-white p-4"
  >
    <h1
      class="text-2xl font-courierbold "
    >
      Great Shop
    </h1>
    <nav
      class="flex gap-10 font-primary text-lg"
    >
      <a
        class=""
        href="/"
      >
        Home
      </a>
      <a
        class=""
        href="/"
      >
        Shop
      </a>
      <a
        class=""
        href="/"
      >
        About
      </a>
    </nav>
    <div>
      <button
        class="bg-cart-logo bg-cover h-8 w-8 border-none pointer-events-auto"
      />
    </div>
  </header>
  <main
    class="bg-sky-img bg-no-repeat bg-cover min-h-[calc(100vh-71px)] opacity-95 flex justify-center items-center content-center gap-10 flex-wrap  "
  >
    <div
      class="max-w-[370px] flex flex-col gap-3 "
    >
      <h1
        class="text-4xl font-bold text-white font-courierbold"
      >
        The Greatest Shop
      </h1>
      <p
        class="text-white font-primary leading-4"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc pulvinar nibh arcu, sed auctor erat maximus sit amet.
      </p>
      <button
        class="bg-white rounded font-bold font-courierbold"
      >
        Go Shopping
      </button>
    </div>
    <div
      class="bg-suit-img bg-no-repeat bg-cover w-[370px] min-h-[320px] rounded-lg shadow-[0px_0px_14px_black]"
    />
  </main>
</div>
`;
